maplibre/src/coords.rs

```diff
@@ -390,70 +392,49 @@ impl WorldTileCoords {
         })
     }
 
-    /// Adopted from [tilebelt](https://github.com/mapbox/tilebelt)
     pub fn build_quad_key(&self) -> Option<Quadkey> {
-        let bounds = ZOOM_BOUNDS[self.z.0 as usize];
-        let x = self.x as u32;
-        let y = self.y as u32;
-
-        if x >= bounds || y >= bounds {
-            return None;
-        }
-
-        let mut key = [ZoomLevel::default(); MAX_ZOOM];
+        // check for out of bound access
+        let TileCoords {
+            x, y, z
+        } = self.into_tile(TileAddressingScheme::XYZ)?;
```

Comment:
Quadkeys should work on WorldTileCoords. Hardcoding here the addressing scheme seems odd

Reply:
The only purpose here is to reuse the bound check code. Quite hacky. I will refactor it a bit to make that clear.